core/src/main/java/org/elasticsearch/action/bulk/BulkProcessor.java

```diff
@@ -348,7 +348,7 @@ public void onFailure(Throwable e) {
                 });
                 success = true;
             } catch (InterruptedException e) {
-                Thread.interrupted();
+                Thread.currentThread().interrupt();
```

Comment:
As this class is used on the client, which we do not control at all, this should be treated as library code. As library we do not know and should also not assume in which environment we are running and how the client wants to respond to interruption. Hence, we should definitely call `Thread.currentThread().interrupt()` here. However, we should restore the flag at the end of the catch block in order to reduce potential side-effects in listeners which we also do not control.